fecgov/fec-cms
# Office page template

## Summary
Office page template. [Mural board of mockups](https://app.mural.co/t/fec4461/m/fec4461/1675268367751/9ddad9551d00a89a468b900e565729c5f1d13b95?sender=0bb5f432-316e-4530-91a5-0470dd7a1e46) for pages that will need to be built with this template
- Resolves #5602

Currently deployed to dev space for testing - Mockups: 
https://dev.fec.gov/about/test-office/
https://dev.fec.gov/admin/pages/12755/view_draft/


### Required reviewers

One UX , one frontend

## Impacted areas of the application

MODIFIED: fec/static/scss/base.scss
MODIFIED: fec/static/scss/components/_contact-items.scss
MODIFIED: fec/home/blocks.py
MODIFIED: fec/home/models.py
MODIFIED: fec/home/templates/blocks/contact-info.html
NEW: fec/static/scss/components/_fec-offices.scss
NEW: fec/home/templates/blocks/simple-document.html
NEW: fec/home/templates/home/office_page.html
NEW: fec/home/migrations/0125_officepage.py


## Screenshots

<img width="1317" alt="Screen Shot 2023-03-15 at 4 05 38 PM" src="https://user-images.githubusercontent.com/5572856/225430066-994ba669-ba86-46b6-a8c7-e574336ce4b4.png">



## How to test
- [x] remove dev deploy task before merge
- `npm run build-sass`
-  Go to Wagtaill admin
- Test new OfficePage template
See examples here:
[Mural board of mockups](https://app.mural.co/t/fec4461/m/fec4461/1675268367751/9ddad9551d00a89a468b900e565729c5f1d13b95?sender=0bb5f432-316e-4530-91a5-0470dd7a1e46)

- Test pages are also created on Dev, such as: 
    -  https://dev.fec.gov/about/test-office/
    -  https://dev.fec.gov/admin/pages/12755/view_draft/
    
**Note:** The only `subpage_types` dict it was added to is `AboutLandingPage`, but it works as a child of the more generic page  types like ResourcePage and CustomPage because they have no  dict restricting  allowable subpage_types`.  If we need it as a child of any of the other other pages with restrictions, we need to add it to their lists.
